generators/preprocessor/pipeline_generator.py

```diff
@@ -0,0 +1,270 @@
+''' This script generates a dataset pipeline script based off a JSON config file.
+ This config  file is generated from user inputs on the frontend. The generated script
+ describes a DatasetPipeline object that can then be used for agnostic model training.
+'''
+import json
+from generators.base_generators import PythonGenerator, JsonGenerator
+
+
+class PipelineGenerator(PythonGenerator):
+    '''Generates a dataset preprocessing pipeline based off a JSON config file.
+    '''
+
+    def __init__(self, pipeline_config="generators/preprocessor/pipeline.json",
+                 mapping_config="generators/preprocessor/variable_map.json"):
+        assert pipeline_config is not None
+
+        # Load config files
+        config_file = open(pipeline_config)  # JSON defining pipeline
+        mapping_file = open(mapping_config)  # JSON defining parameter representations
+        self.pipeline = json.load(config_file)["pipeline"]
+        self.variable_map = json.load(mapping_file)["variable_map"]
+
+        self.dataset = self.pipeline["dataset"]["label"]  # Label for dataset being used
+        self.ops = self.pipeline["operations"]  # Preprocessing performed on the dataset
+
+        super(PipelineGenerator, self).__init__(out="generators/preprocessor/pipeline.py")
+
+    def _indent(self, inc=1):
+        ''' Increments the indent level of the output string.
+        '''
+
+        self.indent_level += inc
+
+    def __unindent(self, dec=1):
+        ''' Decrements the indent level of the output string.
+        '''
+
+        self.indent_level -= dec
+
+    def __spaces(self):
+        '''Returns a string of the correct number of spaces for the current indent level.
+        '''
+
+        return self.indent_str * self.indent_level
+
+    def _write(self, line):
+        ''' Writes a line of code to the file.
+        '''
+
+        self.out.write(self.__spaces() + line)
+
+    def _write_docstring(self, line):
+        ''' Writes a line of docstring to the file.
+            line: Docstring comment without docstring quotes.
+        '''
+
+        self.out.write(self.__spaces() + "'''" + line)
+        self.out.write(self.__spaces() + "'''\n\n")
+
+    def _start_method(self):
+        '''Formats out string to generate a method.
+        '''
+
+        self._indent()
+
+    def _end_method(self):
+        '''Formats out string once method is done.
+        '''
+
+        self._write("\n\n")
+        self.__unindent(2)
```

Comment:
I believe these methods can be deleted because they are already defined in the parent class

Reply:
I fixed this error locally already, for simplicity let's just leave it for the next PR, since I had to make a few more changes to make it work after deleting